src/seml/commands/start.py

```diff
@@ -385,7 +385,10 @@ def start_local_job(
                 temp_dir = os.path.join(SETTINGS.TMP_DIRECTORY, str(uuid.uuid4()))
                 os.mkdir(temp_dir, mode=0o700)
                 load_sources_from_db(exp, collection, to_directory=temp_dir)
-                env = {'PYTHONPATH': f'{temp_dir}:$PYTHONPATH'}
+                # To support `src` layouts we should add the src directory to the PYTHONPATH.
+                # https://packaging.python.org/en/latest/discussions/src-layout-vs-flat-layout/
+                src_temp_dir = os.path.join(temp_dir, 'src')
+                env = {'PYTHONPATH': f'{temp_dir}:{src_temp_dir}:$PYTHONPATH'}
```

Comment:
Is there a rationale why the temp dir has higher priority?

Reply:
If you'd install the package in the src directory but have a folder with the same name in the root directory, the package in root directory would be preferred. Though, there are other settings where this could break down and result in different results. Tldr: I think this is slightly saner, but you can't solve it perfectly given the current infrastructure.